### PR TITLE
feat(onyx-832): Allow user to permanently suppress an artwork recommendation by 'disliking' it

### PR DESCRIPTION
**Pull request by nickskalkin**
This PR resolves [ONYX-832] 

### Description

Enabled a long press context module for "New Works for You" rail and screen. User can dislike an artwork using it and the artwork will be excluded from recommendations. This is behind a feature flag.

<video src="https://github.com/artsy/eigen/assets/3934579/e74d0075-729f-4788-a81b-b1adf6df6fc3" width="400px" />

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

- allow suppressing artwork recommendations (behind FF) - @nickskalkin 

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-832]: https://artsyproduct.atlassian.net/browse/ONYX-832?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- iOS user-facing changes (allow suppressing artwork recommendations (behind FF) - @nickskalkin)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c82890ed806cf71f6e8a779b2487615aaf4c6543
</p>

**Comment by nickskalkin** [comment]
@brainbicycle  we (onyx) want to start using the artwork context menu for suppressing artwork recommendation. I've drafted the implementation, but there are a few things I don't like about it + I know there were some issues with context menu which we need to address first. Can we have a call to discuss them?

File: src/app/Components/ContextMenu/ContextMenuArtwork.tsx [hunk 1]
```diff
@@ -163,7 +184,7 @@ export const ContextMenuArtwork: React.FC<ContextMenuArtworkProps> = ({
 
   const handleContextPress: ContextMenuProps["onPress"] = (event) => {
     if (haptic) {
-      trigger(haptic === true ? "impactLight" : haptic)
+      trigger?.(haptic === true ? "impactLight" : haptic)
```
Comment (MounirDhahri) [inline src/app/Components/ContextMenu/ContextMenuArtwork.tsx:187]: **nitpick:** we can trigger a similar haptic feedback when the artwork disappears ;)

File: src/app/utils/mutations/useDislikeArtwork.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  }
+
+  const newWorksForYouScreenConnection = ConnectionHandler.getConnection(
+    viewer,
+    "NewWorksForYou_artworks"
+  )
+
+  if (newWorksForYouScreenConnection) {
+    ConnectionHandler.deleteNode(newWorksForYouScreenConnection, artworkID)
+  }
```
Comment (MounirDhahri) [inline src/app/utils/mutations/useDislikeArtwork.ts:39]: **idea:** this is a useful logic - could be nice to extract to a method. Something like
```
removeNodeFromConnection({record, connectionName, id})
```

File: src/app/utils/mutations/useDislikeArtwork.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+const DislikeArtworkMutation = graphql`
+  mutation useDislikeArtworkMutation($artworkID: String!) {
+    dislikeArtwork(input: { artworkID: $artworkID, remove: false }) {
+      artwork {
+        id
+        internalID
```
Comment (MounirDhahri) [inline src/app/utils/mutations/useDislikeArtwork.ts:47]: I think we don't need this anymore

**Review by MounirDhahri — APPROVED** [review]
Nice!

**Pull request merged**